src/data-fetcher-loop/data-fetcher-loop.ts

```diff
@@ -77,32 +75,33 @@ export const runDataFetcher = async () => {
     const signedDataFetchIntervalMs = signedDataFetchInterval * 1000;
 
     // Better to log the non-decomposed object to see which URL comes from which chain-provider group.
-    logger.debug('Fetching data from signed APIs.', { signedApiUrls });
+    logger.debug('Signed API URLs.', { signedApiUrls });
     const urls = uniq(
       Object.values(signedApiUrls)
         .map((urlsPerProvider) => Object.values(urlsPerProvider))
         .flat(2)
     );
 
+    const urlCount = urls.length;
+    const staggerTimeMs = signedDataFetchIntervalMs / urlCount;
+    logger.info('Fetching signed data', { urlCount, staggerTimeMs });
     const fetchResults = await Promise.all(
-      urls.map(async (url) =>
-        go(
-          async () => {
-            const signedDataApiResponse = await callSignedApi(
-              url,
-              Math.ceil(signedDataFetchIntervalMs * HTTP_SIGNED_DATA_API_TIMEOUT_MULTIPLIER)
-            );
-            if (!signedDataApiResponse) return;
-
-            for (const signedData of signedDataApiResponse) {
-              saveSignedData(signedData);
-            }
-          },
-          { totalTimeoutMs: signedDataFetchIntervalMs }
-        )
-      )
+      urls.map(async (url, index) => {
```

Comment:
This should be non-throwing so no need to wrap in go anymore.

Reply:
Can following statement throw an error? If can, we also may want to wrap parsing operation be done via promise-utils.
https://github.com/api3dao/airseeker-v2/blob/96924b62c87f0a2a7c6185206cea70fb47772b04/src/data-fetcher-loop/data-fetcher-loop.ts#L63

Reply:
> Can following statement throw an error? If can, we also may want to wrap parsing operation be done via promise-utils.

Yeah. I previously though it can't because we fetch only from our Signed APIs, but later we might be fetching from the ones that are specified in the contract (which will be arbitrary) - so yeah, this should be wrapped.